lib/Notification/Notifier.php

```diff
@@ -159,66 +143,42 @@ public function handleImportFailed(INotification $notification, string $language
 		$l = $this->l10nFactory->get(Application::APP_ID, $languageCode);
 		$param = $notification->getSubjectParameters();
 
-		$targetUser = $this->getUser($param['targetUser']);
 		$notification->setRichSubject($l->t('User import failed'))
-			->setParsedSubject($l->t('User import failed'))
 			->setRichMessage(
 				$l->t('Your import to {user} failed.'),
 				[
-					'user' => [
-						'type' => 'user',
-						'id' => $targetUser->getUID(),
-						'name' => $targetUser->getDisplayName(),
-					],
-				])
-			->setParsedMessage(
-				str_replace(
-					['{user}'],
-					[$targetUser->getDisplayName()],
-					$l->t('Your import to {user} failed.')
-				)
-			);
+					'user' => $this->getUserRichObject($l, $param['targetUser']),
+				]);
 		return $notification;
 	}
 
 	public function handleImportDone(INotification $notification, string $languageCode): INotification {
 		$l = $this->l10nFactory->get(Application::APP_ID, $languageCode);
 		$param = $notification->getSubjectParameters();
 
-		$author = $this->getUser($param['author']);
-		$targetUser = $this->getUser($param['targetUser']);
 		$path = $param['path'];
-		$importFile = $this->getImportFile($author, $path);
+		try {
+			$author = $this->getUser($param['author']);
+			$importFile = $this->getImportFile($author, $path);
```

Comment:
Move these lines out of the try as the catch only handles `$fileRichObject`?

Reply:
`$author` and `$importFile` are not used anywhere else so I find it cleaner to have them in the try to kind of scope them.

Reply:
Also I think getUser may throw if the author does not exist anymore and we do want to go into the catch in this case.